File: 1.Vowel-Balance/main.js
function isBalanced(s) {
  const set = new Set("aeiouAEIOU");
  const n = s.length;
  const half = Math.floor(n / 2);

  let balance = 0;
  for (let i = 0; i < half; i++) {
    if (set.has(s[i])) balance++;
    if (set.has(s[n - i - 1])) balance--;
  }

  return balance === 0;
}

File: 1.Vowel-Balance/main.py
def is_balanced(s):
    vowels = set("aeiouAEIOU")
    n = len(s)
    half = n // 2

    balance = 0

    for i in range(half):
        if s[i] in vowels:
            balance += 1
        if s[n - i - 1] in vowels:
            balance -= 1

    return balance == 0
